### DISCUSSION TITLE
Expose version in PR title when there's multiple variations of the same version

### DISCUSSION THREAD

**Discussion by GreyTeardrop** [renovatebot/renovate#17448] [Request Help]
Hello, Renovate team!

I use `regex` versioning to handle some non-standard Docker tag versioning (`eclipse-temurin:11.0.16_8-jdk`). That works great but has a side effect: if my repository includes several variations of the same dependency (`eclipse-temurin:11.0.16_8-jdk`, `eclipse-temurin:11.0.16_8-jre`), then Renovate won't add the new version to the upgrade PR unless it's a major upgrade. Is there a way I could tweak my versioning definition to let Renovate include the version in the PR title?

This seems to affect Renovate Open Source, Renovate On-Premise, and the Github Renovate app.

#### Reproducer

Repository: https://github.com/GreyTeardrop/renovate-tests/. 

Relevant Renovate configuration: 
```json
{
  "packageRules": [
    {
      "matchDatasources" : ["docker"],
      "matchPackageNames" : ["eclipse-temurin"],
      "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+)\\.(?<patch>\\d+)_(?<build>\\d+))?(-(?<compatibility>.*))?$"
    }
  ]
}
```

PR for the `regex` versioning: [Update eclipse-temurin Docker tag](https://github.com/GreyTeardrop/renovate-tests/pull/5). Oddly enough, PR history indicates the Renovate app created that PR without a version in the title, then it did change the PR title to include the dependency version on August 9, then changed it back to versionless title on August 10.

PR for the `docker` versioning: [Update openjdk Docker tag to v11.0.16](https://github.com/GreyTeardrop/renovate-tests/pull/3)

**Comment by rarkins** [comment]
There's no way to do that currently. PRs include version information in the PR title if either:
- versions are identical, or
- it's a major update and all the same major

**Reply by GreyTeardrop** [reply]
Thank you @rarkins! I've played with adding `valueToVersion` to the `RegExpVersioning` and it seems to work for me, but it's also too generic and might probably have too unpredictable consequences.
```typescript
valueToVersion(version: string): string {
  const parsedVersion = this._parse(version);
  if (!parsedVersion) {
    return version;
  }
  return parsedVersion.release.map(String).join('.');
}
```

```
DRY-RUN: Would create PR: Update eclipse-temurin Docker tag to v11.0.16.8 (repository=GreyTeardrop/renovate-tests, branch=renovate/eclipse-temurin-11.x)
```

**Reply by viceice** [reply]
that would have too many side effects

**Reply by GreyTeardrop** [reply]
I see. I would love to contribute a change to remove this small nuance, but can't think of an obvious approach. The only idea I have is to introduce a new `delegate` versioning that would apply a regex replace to the incoming version `11.0.16_8-jdk`, transform it into `11.0.16.8-jdk` then pass to the `docker` versioning.
